### PR TITLE
Simplifying dependencies

### PR DESCRIPTION
**Pull request by nmerket**
## Pull Request Description

Removes most of the dependency version pins so it'll be able to update more easily. 

## Checklist

Not all may apply

- [x] Code changes (must work)
- [x] ~~Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)~~
- [x] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/ci.yml` as necessary.
- [x] All other unit tests passing
- [x] ~~Update validation for project config yaml file changes~~
- [x] ~~Update existing documentation~~
- [x] Run a small batch run to make sure it all works (local is fine, unless an Eagle specific feature)
- [x] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -57,36 +57,31 @@ def run_tests(self):
...
         'dask[complete]>=2022.10.0',
         'docker',
-        'boto3>=1.10.44',
-        's3fs>=0.4.0,<0.5.0',
-        'fsspec>=0.6.0',
-        'docutils<0.15,>=0.10',
-        'yamale>=2.0',
-        'ruamel.yaml>=0.15.0',
-        'testfixtures',
+        's3fs[boto3]',
```
Comment (nmerket) [inline setup.py:66]: We had to pin s3fs to a really old version so we could have boto3 and s3fs coexist. Turns out [we weren't the only ones with this problem](https://github.com/fsspec/s3fs/issues/357) so they made a way to resolve the conflict, which we're using now. This allows us to use a newer version of s3fs among other things.

File: create_eagle_env.sh [hunk 2]
```diff
@@ -26,13 +26,13 @@ MY_CONDA_PREFIX="$CONDA_ENVS_DIR/$MY_CONDA_ENV_NAME"
...
-source deactivate 
-source activate "$MY_CONDA_PREFIX"
+mamba create -y --prefix "$MY_CONDA_PREFIX" -c conda-forge "python=3.10" "pyarrow" "numpy" "pandas" "dask>=2022.10.0" "distributed" "ruby"
+conda deactivate
+conda activate "$MY_CONDA_PREFIX"
 which pip
 if [ $DEV -eq 1 ]
 then
-    pip install --ignore-installed -e .
+    pip install --no-cache-dir --ignore-installed -e .
```
Comment (nmerket) [inline create_eagle_env.sh:35]: Keeps us from filling up a user's home directory with cached pip wheels. IYKYK.

File: create_eagle_env.sh [hunk 3]
```diff
@@ -26,13 +26,13 @@ MY_CONDA_PREFIX="$CONDA_ENVS_DIR/$MY_CONDA_ENV_NAME"
 echo "Creating $MY_CONDA_PREFIX"
 module load conda
 conda remove -y --prefix "$MY_CONDA_PREFIX" --all
-conda create -y --prefix "$MY_CONDA_PREFIX" -c conda-forge "pyarrow>=7.0.0" "python=3.9" "numpy>=1.20.0" "pandas>=1.0.0,!=1.0.4" "dask>=2022.10.0" "distributed>=2021.5" ruby
-source deactivate 
-source activate "$MY_CONDA_PREFIX"
+mamba create -y --prefix "$MY_CONDA_PREFIX" -c conda-forge "python=3.10" "pyarrow" "numpy" "pandas" "dask>=2022.10.0" "distributed" "ruby"
```
Comment (nmerket) [inline create_eagle_env.sh:29]: Turns out they have mamba now, which makes things a lot faster.

File: create_eagle_env.sh [hunk 4]
```diff
@@ -26,13 +26,13 @@ MY_CONDA_PREFIX="$CONDA_ENVS_DIR/$MY_CONDA_ENV_NAME"
...
+conda deactivate
+conda activate "$MY_CONDA_PREFIX"
```
Comment (nmerket) [inline create_eagle_env.sh:31]: Swtiching to `conda deactivate` which we were getting warnings about.

**Review by nmerket — COMMENTED** [review]
See below for more specifics. In general, I'm allowing the latest versions of things to be installed again. Having so many dependencies was causing later versions of pip to run around in circles trying to find compatible combinations of dependencies.

**Review by rajeee — APPROVED** [review]
Thanks @nmerket. Looks good to me!

**Pull request merged**